soroban-env-host/src/vm/parsed_module.rs

```diff
@@ -0,0 +1,550 @@
+                    costs.n_globals = costs.n_globals.saturating_add(s.count());
+                    for global in s {
+                        let global = host.map_err(global)?;
+                        Self::check_const_expr_simple(&host, &global.init_expr)?;
+                    }
+                }
+                ExportSection(s) => costs.n_exports = costs.n_exports.saturating_add(s.count()),
+                ElementSection(s) => {
+                    costs.n_elem_segments = costs.n_elem_segments.saturating_add(1);
+                    elements = elements.saturating_add(s.count());
+                    for elem in s {
+                        let elem = host.map_err(elem)?;
+                        match elem.kind {
+                            ElementKind::Declared | ElementKind::Passive => (),
+                            ElementKind::Active { offset_expr, .. } => {
+                                Self::check_const_expr_simple(&host, &offset_expr)?
+                            }
+                        }
+                        match elem.items {
+                            ElementItems::Functions(_) => (),
+                            ElementItems::Expressions(_, exprs) => {
+                                for expr in exprs {
+                                    let expr = host.map_err(expr)?;
+                                    Self::check_const_expr_simple(&host, &expr)?;
+                                }
+                            }
+                        }
+                    }
+                }
+                DataSection(s) => {
+                    costs.n_data_segments = costs.n_data_segments.saturating_add(1);
+                    for d in s {
+                        let d = host.map_err(d)?;
+                        if d.data.len() > u32::MAX as usize {
+                            return Err(host.err(
+                                ScErrorType::WasmVm,
+                                ScErrorCode::InvalidInput,
+                                "data segment too large",
+                                &[],
+                            ));
+                        }
+                        data = data.saturating_add(d.data.len() as u32);
+                        match d.kind {
+                            wasmparser::DataKind::Active { offset_expr, .. } => {
+                                Self::check_const_expr_simple(&host, &offset_expr)?
+                            }
+                            wasmparser::DataKind::Passive => (),
+                        }
+                    }
+                }
+                CodeSectionEntry(s) => {
+                    let ops = host.map_err(s.get_operators_reader())?;
+                    for _op in ops {
+                        costs.n_instructions = costs.n_instructions.saturating_add(1);
+                    }
+                }
+            }
+        }
+        if data > costs.n_memory_pages.saturating_mul(0x10000) {
```

Comment:
Where does 0x10000 come from? Maybe use a named constant for clarity?

Reply:
Fixed.